### PR TITLE
Allow specific PInvoke calls for Certs and Domains detectors

### PR DESCRIPTION
**Pull request by nmallick1**
Update API Block config to allow specific PInvoke calls for Certs and Domains detectors.
Also consider EntryPoint in DllImport attribute while evaluating PInvoke calls.

### CODE REVIEW

**Review by yifguoMSFT — APPROVED** [review]

File: src/Diagnostics.Scripts/ApiBlockConfig.json [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
           "MessageToShowWhenBlocked": "PInvoke calls are prohibited."
         }
       ]
+    },
+    {
+      "BlockMatchingRegExPattern": "^(dnsapi.dll|dnsapi)\\b",
+      "MessageToShowWhenBlocked": "",
+      "MethodsToBlock": [
+        {
+          "BlockMatchingRegExPattern": "^(?!(DnsQuery_W|DnsRecordListFree)\\b).*",
```
Comment (hforeste) [inline src/Diagnostics.Scripts/ApiBlockConfig.json:18]: if I understand this correctly, this means block all methods that does not have DnsQuery_W or DnsRecordListFree??

Comment (nmallick1) [reply src/Diagnostics.Scripts/ApiBlockConfig.json:18]: Yes, that is correct. Block call calls other than DnsQuery_W and DnsRecordListFree from dnsapi.dll

File: src/Diagnostics.Scripts/CompilationService/CustomCodeAnalyzers/BlockCallsAnalyzer.cs [hunk 2]
```diff
@@ -106,7 +106,9 @@ private void HandleInvocation(OperationAnalysisContext operationContext)
                         _blockConfig.GetMatchingPInvokeConfig(targetPInvokeDll)?.ForEach(piConfig => {
                             if (piConfig.MethodsToBlock?.Count > 0)
                             {
-                                BlockConfig.GetMatchingBlockMessageList(piConfig.MethodsToBlock, invocation.TargetMethod.Name)?.ForEach(strMessage =>
+                                string pInvokeFunctionName = !string.IsNullOrWhiteSpace(invocation.TargetMethod.GetDllImportData().EntryPointName)? invocation.TargetMethod.GetDllImportData().EntryPointName : invocation.TargetMethod.Name;
```
Comment (michimune) [inline src/Diagnostics.Scripts/CompilationService/CustomCodeAnalyzers/BlockCallsAnalyzer.cs:109]: nit: you could write line like this: string pInvokeFunctionName = invocation.TargetMethod.GetDllImportData().EntryPointName ?? invocation.TargetMethod.Name;

**Review by michimune — APPROVED** [review]

File: src/Diagnostics.Scripts/CompilationService/CustomCodeAnalyzers/BlockCallsAnalyzer.cs (hunk 2 above)
Comment (nmallick1) [reply src/Diagnostics.Scripts/CompilationService/CustomCodeAnalyzers/BlockCallsAnalyzer.cs:109]: @michimune won't ?? check only for NULL and not for whitespace/empty string?

Comment (michimune) [reply src/Diagnostics.Scripts/CompilationService/CustomCodeAnalyzers/BlockCallsAnalyzer.cs:109]: Sure, you are correct. I am going to dismiss my comment.

**Review by hforeste — APPROVED** [review]

**Pull request merged**